pyrevolve/revolve_bot/revolve_module.py

```diff
@@ -265,6 +323,38 @@ def to_sdf(self, tree_depth='', parent_link=None, child_link=None):
         parent_link.append(imu_sensor)
         return visual, collision, imu_sensor
 
+    def count_module_connections(self) -> int:
+        """
+        Count how many connections the module has.
+        Connected TouchSensor and BrickSensor Modules are ignored
+        :return: number of connections
+        """
+        # The CoreModule does not have a parent module
+        return super(CoreModule, self).count_module_connections() - 1
+
+    def is_folding(self) -> bool:
+        # parent slot is not supposed to be set
+        parent_slot = None
+        target_slot = None
+        for slot, child in self.iter_children():
+            # is the child slot occupied?
+            if child is None:
+                continue
+            # ignore TouchSensor and BrickSensor modules
+            if isinstance(child, TouchSensorModule) or \
+                    isinstance(child, BrickSensorModule):
+                continue
+            if parent_slot is None:
+                parent_slot = slot
+            elif target_slot is None:
+                target_slot = slot
+            else:
+                # More than 2 children, this is not a fold
+                return False
+        if parent_slot is None or target_slot is None:
+            return False
+        return parent_slot.opposite() != target_slot
```

Comment:
This will only trigger after executing line 350 is executed, but this does not necessarily need to be opposite right? After that it cannot be changed since you already return that it is False.
Overall very confused by this entire function.